src/wrappers/themis/rust/libthemis-src/src/lib.rs

```diff
@@ -146,14 +148,14 @@ impl Build {
             fs::remove_dir_all(&themis_install_dir).expect("rm -r themis/install");
         }
 
-        // TODO: fix a bug in copy_dir which ignores directory source if it is a symlink
-        copy_dir::copy_dir(&themis_src_dir, &themis_build_dir).expect("cp -r src build");
+        copy_directory(&themis_src_dir, &themis_build_dir).expect("cp -r src/ build");
```

Comment:
why there is a need to copy source code?

Reply:
is there any system call that does the same?

Reply:
Uh... um... now that you mention it... To be honest, I think it is so because I'm lazy and stopped thinking. You comment made me wonder about the reasons and now I believe it can be improved.

Initially I added copying as a quick fix for the build. Cargo forbids build scripts to write anything outside or Rust's build directory (OUT_DIR). By default, running `make install` for Themis puts the build directory into `build` in the current directory. So when `build.rs` builds Themis it results in a 'dirty' source directory and fails the package build. I shrugged and did the first thing that came to my mind: added `copy_dir` call to move the source code into OUT_DIR. It worked so I never thought about it again.

Now I see that Themis Makefile has `BUILD_PATH` parameter which controls the build directory location. Therefore it should not be necessary to copy the source code, we can just set `BUILD_PATH=$OUT_DIR/themis/build` or something and that should work. Thank you for pointing this out!

(And yeah, there's no system call or standard library function to copy a directory. Probably because there are different ways one would like to handle special cases. For example, how to treat symlinks, what do when a directory is copied into itself, etc.)

Reply:
@vixentael,

> regarding `BUILD_PATH` – will you add it in this PR?

Yes. Actually, I need just a few finishing touches and the commit should be ready.